tmd/Topology/analysis.py

```diff
@@ -20,6 +20,67 @@ def sort_ph(ph):
     return np.array(ph)[np.argsort([p[0] - p[1] for p in ph])].tolist()
 
 
+def closest_ph(ph_list, target_extent, method='from_above', return_index=False):
+    """
+    Returns the barcode list that has the maximum extent
+    that is closer to the target_extent according to the selected method.
+
+    method:
+        from_above: smallest maximum extent that is greater or equal than target_extent
+        from_below: biggest maximum extent that is smaller or equal than target_extent
+        nearest: closest by absolute value
+    """
+    n_bars = len(ph_list)
+
+    max_extents = \
+    np.fromiter((max(get_lengths(barcode)) for barcode in ph_list), dtype=np.float)
+
+    sorted_indices = np.argsort(max_extents, kind='mergesort')
+    sorted_extents = max_extents[sorted_indices]
+
+    if method == 'from_above':
+
+        above = np.searchsorted(sorted_extents, target_extent, side='right')
+
+        # if target extent is close to current one, return this instead
+        if above - 1 >= 0 and np.isclose(sorted_extents[above - 1], target_extent):
+            closest_index = above - 1
+        else:
+            closest_index = above
+
+        closest_index = np.clip(closest_index, 0, n_bars - 1)
+
+    elif method == 'from_below':
+
+        below = np.searchsorted(sorted_extents, target_extent, side='left')
+
+        # if target extent is close to current one, return this instead
+        if below < n_bars and np.isclose(sorted_extents[below], target_extent):
+            closest_index = below
+        else:
+            closest_index = below - 1
+
+        closest_index = np.clip(closest_index, 0, n_bars - 1)
+
+    elif method == 'nearest':
+
+        below = np.searchsorted(sorted_extents, target_extent, side="left")
+        pos = np.clip(below, 0, n_bars - 2)
+
+        closest_index = \
+        min((pos, pos + 1), key=lambda i: abs(sorted_extents[i] - target_extent))
+
+    else:
+        raise TypeError('Unknown method {} for closest_ph'.format(method))
+
+    index = sorted_indices[closest_index]
+
+    if return_index:
```

Comment:
I am really not a big fan of optional return value. Bugs that results from them can be hard to track.
I think it is better to always return both.

Reply:
Sure, not a big difference.

Reply:
Actually, I made it so it returns only the index. The list is available a priori anyway.